testsuite/libra-fuzzer/Cargo.toml

```diff
@@ -40,3 +40,7 @@ stats_alloc = "0.1.8"
 [[test]]
 name = "artifacts"
 harness = false
+
+[[bin]]
+name = "investigate"
+path = "bin/investigate.rs"
```

Comment:
why place this in source file in bin? We have two patterns in our code... one where we place everything in src and some cases where we use bin.

Reply:
the file has a `main()` and is meant to be compiled as a binary, not a library.

Reply:
This needs to be inside `src/bin/` directory, not `bin/`. Its completely find to have a main.rs and a lib.rs next to one another and we do this in a few places if i can recall. This particular one is making a binary that is named differently from the crate it lives in so I think its a good thing to place it in `src/bin/investigate.rs`. Also I *think* you should be able to remove this section from the .toml file as cargo should be smart enough to find all files under `src/bin` and turn them into binaries named after the name of the .rs file.

Reply:
> Its completely find to have a main.rs and a lib.rs next to one another and we do this in a few places if i can recall

Maybe I'm biased as Golang doesn't allow you to mix a main with a lib, which makes packaging really clean imo. (So you always have to move your main to a different folder if you're writing a lib.)